addons/mail/static/src/core/common/channel_member_model.js

```diff
@@ -16,12 +16,21 @@ export class ChannelMember extends Record {
         return super.insert(...arguments);
     }
 
+    update() {
+        super.update(...arguments);
+        if (!this.localSeenAssigned) {
+            this.localSeenMessage = this.seen_message_id;
+            this.localSeenAssigned = true;
+        }
+    }
+
```

Comment:
`update()` is not intended to be necessarily called. This function exists just for the sake of being a syntatic sugar for `Object.assign()`. So putting custom behaviour on `update` of this specific model is not a good idea.

Instead, what you could do is made another `initialLocalSeenMessage` field, whose compute relies on the 1st set value of `seen_message_id`...
... Or avoiding setting any value until this needs to be used the first time. There's likely a need to distinguish "`localSeenMessage` is set to nothing" vs. "`localSeenMessage` is unset". As relational field are currently designed, there's no distinct value between them so a relational field value would not be enough.
The `localSeenAssigned` here looked a bit like a field that could be dedicated for that purpose.

Reply:
We need to set the value: if we mark the first message of a chat as unread, the  `seen_message_id` would be undefined. In such a case, we won't have any solution to mark the local seen message as assigned (ideally, we should rely on `onAdd`/`onUpdate` but with the current implementation, the local seen will be unchanged (undefined => undefined). So we have to consider that local seen message with an undefined value means the first message of a chat is unread.